src/main/java/org/apache/mesos/scheduler/plan/DefaultPhase.java

```diff
@@ -12,7 +15,7 @@
  * their ordering. It is an immutable class that can be constructed either
  * directly, or using a fluent style builder.
  */
-public class DefaultPhase implements Phase {
+public class DefaultPhase extends ChainedObserver implements Phase, Observer {
```

Comment:
Do you need to write `implements Observer` if you already extend the `ChainedObserver`?